/*
 *  Copyright (c) 2025 The Android Open Source Project
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at:
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

syntax = "proto3";

package bluetooth_hal.config.proto;

message HalConfig {
  optional bool fast_download_enabled = 1;
  optional bool sar_backoff_high_resolution_enabled = 2;
  optional int32 reg_on_delay_ms = 3;
  optional string uart_device_port = 4;
  repeated int32 transport_type_priority = 5;
  optional bool accelerated_bt_on_enabled = 6;
  optional bool thread_dispatcher_enabled = 7;
  optional bool bt_power_controlled_by_lpp = 8;
  repeated string hw_stages_without_lpp_control_bt_power_pin = 9;
  repeated string unsupported_hw_stages = 10;
  optional int32 vendor_transport_crash_interval_sec = 12;
  optional bool hp_uart_skip_suspend_enabled = 13;
  optional bool energy_controller_logging_enabled = 14;
  optional bool self_restart_recovery_enabled = 15;
  optional bool ble_non_connection_sar_enabled = 16;
  optional int32 kernel_rx_wakelock_time_ms = 17;
  optional bool low_power_mode_enabled = 18;
  optional string bqr_event_mask = 19;
  optional string ldac_quality_mode = 20;
  optional int32 transport_fallback_type = 21;
  optional string lpm_enable_proc_node = 22;
  optional string lpm_waking_proc_node = 23;
  optional string lpm_wakelock_ctrl_proc_node = 24;
  optional string rfkill_folder_prefix = 25;
  optional string rfkill_type_bluetooth = 26;
}
